crates/holochain/src/conductor/conductor.rs

```diff
@@ -1079,6 +1079,21 @@ mod app_impls {
 
             let cells_to_create = ops.cells_to_create();
 
+            // check if cells_to_create contains a cell identical to an existing one
+            let state = self.get_state().await?;
+            let mut all_cells = state
+                .installed_apps()
+                .values()
+                .flat_map(|app| app.all_cells());
+            let maybe_duplicate_cell_id = cells_to_create
+                .iter()
+                .find(|(cell_id, _)| all_cells.any(|existing_cell_id| existing_cell_id == cell_id));
```

Comment:
Might be better performance, maybe only do this if you were gonna make other changes already, but this should be fine too since `cells_to_create` should be small.

```suggestion
            let mut all_cells: HashSet<_> = state
                .installed_apps()
                .values()
                .flat_map(|app| app.all_cells())
                .collect();
            let maybe_duplicate_cell_id = cells_to_create
                .iter()
                .find(|(cell_id, _)| all_cells.contains(cell_id));
```

Reply:
Lovely, thanks for that! I'm learning these ways of using iterators and the like.